[IMP] Field_Service: Creating Tasks (#10176)
odoo/documentation: content/applications/services/field_service/creating_tasks.rst

============================
Creating Field Service Tasks
============================

Creating tasks in **Field Service** from a sales order or a helpdesk ticket is faster and easier
since these already contain a lot of information that will be imported automatically into your field
service task, but it is also possible to create one manually from scratch.

Manually
========

 #. Go to :guilabel:`Field Service`
 #. Click on :guilabel:`New`
 #. Fill in the :guilabel:`Task Title` and :guilabel:`Customer` (all other fields are optional),
    then click :guilabel:`Save`

> Task configuration

Reviewer: This part is actually already described in the Project doc so I think it would be best to just refer to that doc like I suggested above, and remove all of this so we don't have duplicate content. From what I can see, only the Worksheet template field is field service-specific but it's described in the Worksheets doc so we're okay. I was also wondering about Under warranty, but 1) I can't find it and 2) there's a good chance it would be sales-related as well

 - :guilabel:`Project`: the project that this task belongs to.
 - :guilabel:`Worksheet Template`: the worksheet template you want to use for this task.
 - :guilabel:`Assignees`: the person(s) in charge of handling the work on this task.
 - :guilabel:`Tags`: custom labels allowing to categorize and filter your tasks.
 - :guilabel:`Customer`: the person or company that will be billed for this task. This field only
   appears in tasks that belong to billable projects.
 - :guilabel:`Contact Number`: your customer's phone number
 - :guilabel:`Sales Order Item`: this can be either the sales order that was used to create this
   task, or a sales order that was linked to this task manually. This field only appears in tasks
   linked to billable projects.
 - :guilabel:`Under Warranty` check box: if ticked, the time and materials used for this task will
   not be billed to the customer. However, the inventory of consumed materials will still be
   updated.
 - :guilabel:`Planned Date`: the date you want to schedule for this field service.
 - :guilabel:`Allocated Time`: the amount of time that the work on this task is expected to last,
   tracked by timesheets.

> From a sales order

Reviewer: ```suggestion
Creating field service tasks from a sales order
```
or whatever fits what you chose for the previous heading

==================

Allowing your sales team to create field service tasks offers a seamless experience for your
customers. They can receive a quotation they first have to approve before the work even starts. To

> do this, you first need to create a specific product that you can sell as part of a field service.

Reviewer: - This is a little bit too marketing-y I think: see this for a better example https://www.odoo.com/documentation/master/applications/services/project/tasks/task_creation.html#creating-tasks-from-an-email-alias 
- And since the content is a duplicate from this https://www.odoo.com/documentation/master/applications/sales/sales/invoicing/time_materials.html#service-product-configuration (and we try to avoid that) but we still want to mention somewhere here that you can create FS tasks from a sales order, we could rephrase this whole section to only keep: 1) the fact we can create FS tasks from a sales order 2) that you need a "service" product (add a ref to this as well https://www.odoo.com/documentation/master/applications/sales/sales/invoicing/time_materials.html#service-product-configuration and keep the screenshot! I think it's useful) and 3) you could keep lines 63-65 to mention the process in general.
- Let's discuss it, it's getting complicated xD

Go to :menuselection:`Field Service --> Configuration --> Products` and create or edit a product,
and go to the :guilabel:`General Information` tab.

#. Select :guilabel:`Service` as :guilabel:`Product Type`.
#. Select :guilabel:`Task` in the :guilabel:`Create on Order` drop down menu.
#. Select :guilabel:`Field Service` in the :guilabel:`Project` drop down menu.
#. If you use them, select a :doc:`worksheet <worksheets>` in the :guilabel:`Worksheet Template`
   drop down menu.
#. Select :guilabel:`Based on Timesheets` or any other option in the :guilabel:`Invoicing Policy`
   drop down menu and save.

.. image:: creating_tasks/product-configuration.png
   :alt: Product configuration to create tasks from sales orders in Odoo Field Service

> From the :doc:`Sales <../../sales/sales>` app, create a quotation with the product and confirm it. A
> task is automatically set up under your Field Service project. It is directly accessible from the
> sales order as a smart button.

Reviewer: ```suggestion
When you create a quotation with the service and confirm it, a
task is automatically created in your Field Service project. Click the :guilabel:`Tasks` smart button to acess it from the sales order.
```

> From helpdesk tickets

Reviewer: ```suggestion
Creating field service tasks from helpdesk tickets
```
(same as above, change the wording accordingly)

=====================

> The integration with the :doc:`Helpdesk <../helpdesk>` app lets your helpdesk team manage
> field service requests directly. Planning field service tasks from tickets speeds up your processes.

Reviewer: Same as above, little marketing-y. Instead I would do something like this: https://www.odoo.com/documentation/master/applications/services/project/tasks/task_creation.html#creating-tasks-from-a-website-form 
```suggestion
If you have the Helpdesk app installed, you can :ref:`create field service tasks from a helpdesk ticket <helpdesk/field>`.
```
to link to this doc https://www.odoo.com/documentation/master/applications/services/helpdesk/advanced/after_sales.html#create-field-service-task-from-a-ticket

Configure the helpdesk team

> ---------------------------

Reviewer: Same as above but with this doc: https://www.odoo.com/documentation/master/applications/services/helpdesk/advanced/after_sales.html#create-field-service-task-from-a-ticket
and in this cas, I don't know if we need much else so I would remove this section altogether? Let's discuss it!
